Add an option to return all the sites individually if site_name is blank

raysapida opened the pull request.

raysapida committed:
Add an option to return all the sites individually if site_name is blank

msalo3 commented:
Oh cool - so we can have 1 workflow to grab all sites at once?

msalo3 commented on lib/qbwc/request/inventorywithsites.rb:
```diff
@@ -12,6 +12,14 @@ def polling_current_items_xml(params, config)
           timestamp = params['quickbooks_since']
           session_id = Persistence::Session.save(config, 'polling' => timestamp)
 
+          if site_name(params) == ''
```
what if we just omit site_name?

raysapida replied:
Hmm, it would probably go to the else branch and try to just look for site names with an empty string. Nice catch! I'll add a check to see if the `quickbooks_site` is there at all

raysapida commented:
Yes, it will return them with their associated sites in each inventory

raysapida committed:
Include case where `quickbooks_site` isn't given

- Without this check, it will try to filter the site name by nil and
probably set it to an empty string when its interpolated
